src/vscode-dts/vscode.d.ts

```diff
@@ -7733,6 +7733,19 @@ declare module 'vscode' {
 		 * https://invisible-island.net/xterm/ctlseqs/ctlseqs.html
 		 */
 		readonly isInteractedWith: boolean;
+
+		/**
+		 * The shell type of the {@link Terminal}.
+		 * Examples of some of the identifiable shells include: bash, fish, node, pwsh, python, zsh, etc.
+		 *
+		 * {@link Terminal}'s shell may change throughout its lifetime, such as when user launches bash from zsh.
+		 * Shell type may be undefined if it outside of internal TerminalShellType that is defined in
+		 * https://github.com/microsoft/vscode/blob/main/src/vs/platform/terminal/common/terminal.ts
```

Comment:
We shouldn't ever use a link like this, that file could be moved and it would 404, the content could be moved elsewhere, etc.